File: 250619/k번째로 신기한 문자열/kth-special-string.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        String t = sc.next();
        String[] words = new String[n];
        for (int i = 0; i < n; i++) {
            words[i] = sc.next();
        }
        // Please write your code here.

        int idx = 0;
        String[] selectedWords = new String[n];

        for (int i = 0; i < n; i++) {
            if (t.equals(words[i].substring(0,t.length()))) {
                selectedWords[idx++] = words[i];
            }
        }

        String[] fixedSelectedWords = new String[idx];
        for (int i = 0; i < idx; i++) 
            fixedSelectedWords[i] = selectedWords[i];

        Arrays.sort(fixedSelectedWords);

        System.out.print(fixedSelectedWords[k-1]);
    }
}

File: 250923/k번째로 신기한 문자열/kth-special-string.py
n, k, t = input().split()
n, k = int(n), int(k)
str1 = [input() for _ in range(n)]

# Please write your code here.

def isFirstT(l) :
    return l[0] == T 

str1 = list(filter(isFirstT,str1))

str1.sort()

print(str1[k-1])